sensorprocess/sensorprocess.go

```diff
@@ -77,47 +77,79 @@ func (config *Config) StartLidar(
 
 // addLidarReading adds a lidar reading to the cartofacade.
 func (config *Config) addLidarReading(ctx context.Context) bool {
-	/*
-	 when the lidar data rate msec is non-zero, we assume the user wants to be in "online"
-	 mode and ensure the most recent scan gets processed by cartographer. If data rate msec
-	 is zero we process every scan in order
-	*/
+	// Process data in online or offline mode determined by the lidar's data rate
```

Comment:
Would it be possible to pull out some of the logic to get the next reading/timestamps. I think this function is getting increasingly large and sort of hard to follow

Reply:
Done